pkg/api/api_test.go

```diff
@@ -340,48 +342,212 @@ func TestFqNameFilter(t *testing.T) {
 	}
 }
 
-func TestFieldFilter(t *testing.T) {
+func TestFieldSelector(t *testing.T) {
 	e := setupTestcase(t)
 
 	createObj(t, e, "domain", `{"name": "default"}`)
-	_, projectID := createObj(t, e, "project", `{
-		"name": "juniper", 
+	status, projectID := createObj(t, e, "project", `{
 		"fq_name": ["default", "juniper"], 
 		"display_name": "Juniper Networks"}`)
-	_, deviceID := createObj(t, e, "device", `{
-		"name": "junos",
-		"fq_name": ["default", "juniper", "junos"],
+	require.Equal(t, http.StatusCreated, status)
+	status, device_familyID := createObj(t, e, "device_family", `{
```

Comment:
In CSO, I think some of the entities are using '-' and some with '_', like for this one it is 'device-family', for fmpm I think they are using alert_status_object.

Reply:
Yes, I noticed it also when I was comparing with ems-central vnc APIs. I will open an issue to track it. To fix it, we need to add a method "yangModelName()" in `gen.go` that would return the entity name with `-` and this method needs to be used in `route.go`.